packages/core/src/http-client/http-client.ts

```diff
@@ -201,11 +201,8 @@ function executeWithAxios(request: HttpRequest): Promise<HttpResponse> {
   return axios.request(mergeRequestWithAxiosDefaults(request));
 }
 
-function executeWithAxiosReturnRequestAndResponse(request: HttpRequest): Promise<HttpRequestAndResponse> {
-  const merged = mergeRequestWithAxiosDefaults(request);
-  return axios.request(merged).then(
-    res => ({ request: merged, response: res })
-  );
+function executeWithAxiosReturnAxiosResponse(request: HttpRequest): Promise<AxiosResponse> {
```

Comment:
I am not a fan of this function. It is an exact duplication of the function above and the only difference is, that we are using a different return type. I don't think that the name `executeHttpRequestReturnAxiosResponse` should be used in the public API.
In my opinion we should consider one of two options:
1. Either enrich the `HttpResponse` type with some more information. We could use [this](https://github.com/axios/axios#response-schema) as a basis. The downside here is, that this is not possible without breaking changes. Therefore the added properties would have to be optional or we would have to think of something different.
2. Use `AxiosResponse` in executeWithAxios directly. This should work OOB because `AxiosResponse` is compatible with `HttpResponse`. I am not a big fan of using another tool's API in our own, because they would have to install axios to reference the type.

I don't have a strong opinion here, but maybe @FrankEssenberger has something to add? And maybe @mr-flannery might have some feedback as well?

Reply:
As discussed with Marika. I would go with the option 1 and add an optional property `request` to our interface `HttpResponse`.
- no breaking changes
- since it's compatible with `AxiosReponse`, one can cast it to `AxiosResponse` on demand
- the ugly function will disappear, which does not do anything but having a different return type.